controllers/argocd/util.go

```diff
@@ -1879,3 +1874,168 @@ func appendUniqueArgs(cmd []string, extraArgs []string) []string {
+				return false
+			}
+			valOld, ok := e.ObjectOld.(*argoproj.ArgoCD)
+			if !ok {
+				return false
+			}
+
+			k8sClient, err := initK8sClient()
+			if err != nil {
+				return false
+			}
+
+			// Check if namespaceManagement has changed
+			if !reflect.DeepEqual(valOld.Spec.NamespaceManagement, valNew.Spec.NamespaceManagement) {
+				fmt.Println("NamespaceManagement field changed, triggering reconciliation")
+
+				// Convert old and new NamespaceManagement lists into maps for easy lookup
+				oldNamespaces := make(map[string]bool)
+				newNamespaces := make(map[string]bool)
+				oldNamespaceSettings := make(map[string]bool) // Track allowManagedBy values
+				newNamespaceSettings := make(map[string]bool)
+
+				for _, ns := range valOld.Spec.NamespaceManagement {
+					oldNamespaces[ns.Name] = true
+					oldNamespaceSettings[ns.Name] = ns.AllowManagedBy // Store old setting
+				}
+				for _, ns := range valNew.Spec.NamespaceManagement {
+					newNamespaces[ns.Name] = true
+					newNamespaceSettings[ns.Name] = ns.AllowManagedBy // Store new setting
+				}
+
+				// Find namespaces that were removed or had allowManagedBy changed
+				namespacesToDelete := []string{}
+
+				if len(oldNamespaces) > 0 {
+					for ns := range oldNamespaces {
+						if !newNamespaces[ns] { // Namespace was removed
+							namespacesToDelete = append(namespacesToDelete, ns)
+						} else if oldNamespaceSettings[ns] != newNamespaceSettings[ns] { // allowManagedBy changed
+							namespacesToDelete = append(namespacesToDelete, ns)
+						}
+					}
+				}
+
+				if len(namespacesToDelete) > 0 {
+					for _, namespace := range namespacesToDelete {
+						// Delete RBACs for the removed namespace
+						if err := deleteRBACsForNamespace(namespace, k8sClient); err != nil {
+							log.Error(err, fmt.Sprintf("Failed to delete RBACs for namespace: %s", namespace))
+						} else {
+							log.Info(fmt.Sprintf("Successfully removed the RBACs for namespace: %s", namespace))
+						}
+
+						// Remove namespace from cluster secret
+						if err = deleteManagedNamespaceFromClusterSecret(valOld.GetNamespace(), namespace, k8sClient); err != nil {
+							log.Error(err, fmt.Sprintf("Unable to delete namespace %s from cluster secret", namespace))
+						} else {
+							log.Info(fmt.Sprintf("Successfully deleted namespace %s from cluster secret", namespace))
+						}
+					}
+				}
+			}
+			return true
+		},
+	}
+}
+
+// updateStatusConditionOfArgoCD calls Set Condition of NamespaceManagement status
+func updateStatusConditionOfNamespaceManagement(ctx context.Context, condition metav1.Condition, cr *argoproj.NamespaceManagement, k8sClient client.Client, log logr.Logger) error {
+	changed, newConditions := insertOrUpdateConditionsInSlice(condition, cr.Status.Conditions)
+
+	if changed {
+		// get the latest version of namespacemanagement before updating
+		if err := k8sClient.Get(ctx, types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, cr); err != nil {
+			return err
```

Comment:
Here you can add:
```
if apierrors.IsNotfound(err) {
    return nil
}
```
If the `NamespaceManagement` no longer exists, then we can just `return nil` since there is no more work to do. (I am fixing this bug in another part of the code with a PR)